contribs/drt/src/main/java/org/matsim/contrib/drt/analysis/zonal/ZonalDemandAggregator.java

```diff
@@ -37,63 +37,7 @@
 /**
  * @author jbischoff
  */
-public class ZonalDemandAggregator implements PersonDepartureEventHandler {
+public interface ZonalDemandAggregator {
 
-	private final DrtZonalSystem zonalSystem;
-	private final String mode;
-	private final int timeBinSize;
-	private final Map<Double, Map<String, MutableInt>> departures = new HashMap<>();
-	private final Map<Double, Map<String, MutableInt>> previousIterationDepartures = new HashMap<>();
-
-	public ZonalDemandAggregator(EventsManager events, DrtZonalSystem zonalSystem, DrtConfigGroup drtCfg) {
-		this.zonalSystem = zonalSystem;
-		mode = drtCfg.getMode();
-		timeBinSize = drtCfg.getMinCostFlowRebalancing().get().getInterval();
-		events.addHandler(this);
-	}
-
-	@Override
-	public void reset(int iteration) {
-		previousIterationDepartures.clear();
-		previousIterationDepartures.putAll(departures);
-		departures.clear();
-		prepareZones();
-	}
-
-	@Override
-	public void handleEvent(PersonDepartureEvent event) {
-		if (event.getLegMode().equals(mode)) {
-			Double bin = getBinForTime(event.getTime());
-
-			String zoneId = zonalSystem.getZoneForLinkId(event.getLinkId());
-			if (zoneId == null) {
-				Logger.getLogger(getClass()).error("No zone found for linkId " + event.getLinkId().toString());
-				return;
-			}
-			if (departures.containsKey(bin)) {
-				this.departures.get(bin).get(zoneId).increment();
-			} else
-				Logger.getLogger(getClass())
-						.error("Time " + Time.writeTime(event.getTime()) + " / bin " + bin + " is out of boundary");
-		}
-	}
-
-	private void prepareZones() {
-		for (int i = 0; i < (3600 / timeBinSize) * 36; i++) {
-			Map<String, MutableInt> zonesPerSlot = new HashMap<>();
-			for (String zone : zonalSystem.getZones().keySet()) {
-				zonesPerSlot.put(zone, new MutableInt());
-			}
-			departures.put(Double.valueOf(i), zonesPerSlot);
-		}
-	}
-
-	private Double getBinForTime(double time) {
-		return Math.floor(time / timeBinSize);
-	}
-
-	public Map<String, MutableInt> getExpectedDemandForTimeBin(double time) {
-		Double bin = getBinForTime(time);
-		return previousIterationDepartures.getOrDefault(bin, Collections.emptyMap());
-	}
+	public Map<String, MutableInt> getExpectedDemandForTimeBin(double time);
```

Comment:
To avoid exposing a mutable state, I suggest returning a simple getter function:
`ToIntFunction<String> getExpectedDemandForTimeBin(double time);`